lib/api_repo.js

```diff
@@ -305,14 +305,18 @@ ApiRepo.prototype.buildGaxPackages =
             var pkgName = that.pkgPrefix + cleanName;
             var shortName = name.split('/').slice(-1)[0];
             var titleName = shortName[0].toUpperCase() + shortName.slice(1);
+            var lastDirNameInGaxDir = that.gaxDir.split('/').slice(-1)[0];
             opts.packageInfo.api.simplename = cleanName;
             opts.packageInfo.api.path = cleanName.replace(/-/g, '/');
-            if (l === 'nodejs' && cleanName.indexOf('google-') === 0) {
-              opts.packageInfo.api.name = '@google-cloud/' +
-                  cleanName.replace(/^google-(cloud-)?/, '');
+
+            // For NodeJS and Ruby, we build the api name based on the last dir name from
+            // the full path of gaxDir (i.e. final_repo_dir)
+            if ((l === 'nodejs' || l === 'ruby') && cleanName.startsWith('google-')) {
+              opts.packageInfo.api.name = (l === 'nodejs') ? '@google-cloud/' + lastDirNameInGaxDir : lastDirNameInGaxDir;
```

Comment:
I think the lint checker will raise a warning here. Please make sure not exceeding 80-characters line limit.